posts/inside-rust/2019-10-11-AsyncAwait-Not-Send-Error-Improvements.md

```diff
@@ -121,9 +121,25 @@ async fn bar(x: &Mutex<u32>) {
 ```
 
 ```rust
-enum BarGenerator {
-    Unresumed { x: &Mutex<u32>, g: MutexGuard<'_, u32>, },
-    Suspend0 { x: &Mutex<u32>, g: MutexGuard<'_, u32>, }
+enum BarAsyncFnGenerator {
```

Comment:
I prefer the enum name as it was, it isn't that important but it's just a bit simpler and I don't think it was ambiguous what "Bar" was referring to previously.

```suggestion
enum BarGenerator {
```